requirements.txt

```diff
@@ -1,8 +1,11 @@
 numpy
-Cython
+Cython==0.29.36
```

Comment:
Why not `Cython<=3.0.3`?

Reply:
the last known working version of the NGIAB-CloudInfra docker image (5 months ago) was using cython 0.29.28. I don't understand the outputs of t-route enough to verify that 3.0.2 works properly so I didn't want to jump major versions.
If someones able to verify 3.0.2 works then it'd be great to use a more recent version :)

Reply:
I double-checked in both versions and the `NextGen` still has a segmentation fault error. So, I think we should keep `Cython<=3.0.3` and then try to solve the error.

Reply:
@TrupeshKumarPatel, as a general rule of thumb you we would probably want to pin its as `Cython~=3.0.3` until the `Cython==3.0.4` issue is resolved.

Reply:
I created #675 to track the `Cython` issue. In short, until `Cython==3.0.5` has been released and verified to build t-route correctly, I would suggest we pin `Cython~=3.0.3`.

Reply:
I agreed with @aaraney to keep `Cython<3.0.4`. 
@JoshCu, if you agree, could you make changes accordingly?

Reply:
Apologies for the delay, I've changed the requirements.txt to `Cython<3.0.4`